fieldservice_account/models/account_move.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (C) 2018, Open Source Integrators
+# Copyright 2019 Akretion <raphael.reverdy@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    fsm_order_ids = fields.Many2many(
+        "fsm.order",
+        compute="_compute_fsm_order_ids",
+        string="Field Service orders associated to this invoice",
+    )
+    fsm_order_count = fields.Integer(
+        string="FSM Orders", compute="_compute_fsm_order_ids"
+    )
+
+    @api.depends("line_ids")
+    def _compute_fsm_order_ids(self):
+        for order in self:
+            orders = self.env["fsm.order"].search(
+                [("invoice_lines", "in", order.line_ids.ids)]
+            )
+            order.fsm_order_ids = orders
+            order.fsm_order_count = len(order.fsm_order_ids)
+
+    def action_view_fsm_orders(self):
+        action = self.env.ref("fieldservice.action_fsm_dash_order").read()[0]
```

Comment:
use the odoo method _for_xml_id it is safer
```suggestion
        action = self.env["ir.actions.act_window"]._for_xml_id(
            "fieldservice.action_fsm_dash_order"
        )
```